File: packages/react-native-messaging-core/ios/Sources/Models/Provider.swift
import Foundation
import NablaMessagingCore

extension Provider {
    var dictionaryRepresentation: [String: Any] {
        var result = [String: Any]()
        result["id"] = id
        result["avatarURL"] = avatarURL
        result["prefix"] = prefix
        result["firstName"] = firstName
        result["lastName"] = lastName
        return result
    }
}

File: packages/react-native-messaging-core/android/src/main/java/com/nabla/sdk/reactnative/messaging/core/models/Provider.kt
package com.nabla.sdk.reactnative.messaging.core.models

import com.facebook.react.bridge.Arguments
import com.facebook.react.bridge.ReadableMap
import com.nabla.sdk.core.domain.entity.Provider

internal fun Provider.toMap(): ReadableMap = Arguments.createMap().apply {
    putString("id", id.toString())
    putString("prefix", prefix)
    putString("firstName", firstName)
    putString("lastName", lastName)
    avatar?.let {  putString("avatarURL", it.url.toString())}
}
